backend/modules/parsers/parser.py

```diff
@@ -54,6 +54,29 @@ async def get_chunks(
         pass
 
 
+def get_parser_for_extension_with_cache(
+    extension: str, parsers_map: Dict[str, Any], cache: dict, *args, **kwargs
+):
+    """
+    Retrieves a parser for the given file extension using a cache.
+
+    Args:
+        extension (str): The file extension to get a parser for.
+        parsers_map (Dict[str, Any]): Mapping of file extensions to parser configurations.
+        cache (dict): A dictionary to cache parsers.
+        *args: Variable length argument list.
+        **kwargs: Arbitrary keyword arguments.
+
+    Returns:
+        BaseParser: The parser for the given extension, either from cache or newly created.
+    """
+    if extension not in cache:
+        cache[extension] = get_parser_for_extension(
+            extension, parsers_map, *args, **kwargs
+        )
+    return cache[extension]
```

Comment:
one thing that is not clear to me is, how will two parsers with same extension but different config params will be cached ? Usecase being when the collection has multiple datasources, each data sources uses the same parser but with different config.

Reply:
You are right. I've changed the cache key in the latest commits to include parser_config.